File: app/src/main/java/se/hactar/movieregister/MovieItemClickListener.java
package se.hactar.movieregister;

import android.content.Intent;
import android.net.Uri;
import android.view.View;
import android.widget.AdapterView;

import se.hactar.movieregister.data.Movie;

class MovieItemClickListener implements AdapterView.OnItemClickListener {
    @Override
    public void onItemClick(final AdapterView<?> parent, final View view, final int position, final long id) {
        Movie movie = (Movie) parent.getAdapter().getItem(position);
        if (movie.getId().length() == 0) {
            // TODO: Dialog that say IMDB ID is missing.
            return;
        }
        String url = "http://www.imdb.com/title/" + movie.getId() + "/";
        Intent intent = new Intent(Intent.ACTION_VIEW);
        intent.setData(Uri.parse(url));
        parent.getContext().startActivity(intent);
    }
}

File: app/src/main/kotlin/se/hactar/movieregister/MovieItemClickListener.kt
package se.hactar.movieregister

import android.content.Intent
import android.net.Uri
import android.view.View
import android.widget.AdapterView
import se.hactar.movieregister.data.Movie

/**
 * Created by dilbert on 2015-10-14.
 */
class MovieItemClickListener : AdapterView.OnItemClickListener {
    override fun onItemClick(parent: AdapterView<*>?, view: View?, position: Int, id: Long) {
        val movie = parent!!.adapter.getItem(position) as Movie
        if (movie.imdbId.length == 0) {
            // TODO: Dialog that say IMDB ID is missing.
            return
        }
        val url = "http://www.imdb.com/title/" + movie.imdbId + "/"
        val intent = Intent(Intent.ACTION_VIEW)
        intent.setData(Uri.parse(url))
        parent.context.startActivity(intent)
    }
}